### PR TITLE
Fix WRFPlus compilation issue

### PR DESCRIPTION
**Pull request by liujake**
TYPE: bug fix

KEYWORDS: wrfplus, TL/AD code

SOURCE: internal

DESCRIPTION OF CHANGES:
Two recent WRF model code changes break the wrfplus compilation:
1. ESMF name change 
This modification allowed the CTSM development to use standard ESMF without conflicting 
with WRF _old_ ESMF library namespace.
Commit 8af53855b837, PR #1066 "Enabling WRF build with ESMF library"
2. module_diffusion_em_ad.F, 3d scale aware TKE (one of these, take your pick)
Commit d059afe1e646, PR #972 "Add a scale-adaptive 3DTKE parameterization scheme as a subgrid mixing option
Commit 6d95883111f5, PR #1012 "SMS3dTKE: Remove broken packaging for km_opt=5"
Commit 94e5ed97b69, PR #1013 "Add missing "IF km_opt==5" tests for diag computations"
Commit dd0c2094fc16, PR #1026 "Fixing the problem that km_opt=5 breaks km_opt=2 and cleaning codes"

LIST OF MODIFIED FILES: 
M       share/mediation_integrate.F
M       wrftladj/module_diffusion_em_ad.F

TESTS CONDUCTED: wrfplus code compiles now.

### CODE REVIEW

File: main/depend.common [hunk 1]
```diff
@@ -701,6 +701,7 @@ module_radiation_driver.o: \
                 module_ra_rrtmg_lwk.o \
                 module_ra_rrtmg_swk.o \
 		module_ra_cam.o \
+		module_ra_farms.o \
```
Comment (davegill) [inline main/depend.common]: @liujake 
Jake,
There is a separate PR for this mod, which is a more general fix than just for DA.
Please remove it.

Comment (liujake) [reply main/depend.common]: I thought this should disappear after you merge your PR. Apparently not.

Comment (davegill) [reply main/depend.common]: No such luck.
I can easily remove this line via github, if you like.

Comment (liujake) [reply main/depend.common]: Removed it online.

**Comment by davegill** [comment]
@jjguerrette 
Likely will be approving this. I wanted to chime in that I am OK with this PR.
 
1. I asked Jake to remove a redundant mod that would have caused a conflict - it is gone. 
2. The modification inside of `share/mediation_integrate.F` is inside of a wrfplus ifdef, so no impact on anything other than DA.
3. The mod does what Jake wanted - make DA build.

**Pull request merged**